Computing the roots of a polynomial in a field extension of `QQ(t)` yields an `AssertionError`

The following code
```
Qt, t = rational_function_field(QQ, :t)
Qtx, x = Qt[:x]
f = 5*x^12 + 380*x^9 - 240*x^6 - 1600*x^3 - 256
g = x^12 + 9500*x^9 - 750000*x^6 - 625000000*x^3 - 12500000000
K = splitting_field(g)
roots(K, f)
```
outputs an `AssertionError: isone(fN.unit)`.
The full output is given below:
```
julia> Qt, t = rational_function_field(QQ, :t)
(Rational function field over QQ, t)

julia> Qtx, x = Qt[:x]
(Univariate polynomial ring in x over Qt, x)

julia> f = 5*x^12 + 380*x^9 - 240*x^6 - 1600*x^3 - 256
5*x^12 + 380*x^9 - 240*x^6 - 1600*x^3 - 256

julia> g = x^12 + 9500*x^9 - 750000*x^6 - 625000000*x^3 - 12500000000
x^12 + 9500*x^9 - 750000*x^6 - 625000000*x^3 - 12500000000

julia> K = splitting_field(g)
Function Field over QQ with defining polynomial b^24 + 189000*b^21 + 10806750000*b^18 + 61053750000000*b^15 + 198497587500000000*b^12 + 106534237500000000000*b^9 + 11899788215625000000000000*b^6 + 23433226593750000000000000000*b^3 + 18847970253281250000000000000000

julia> roots(K, f)
ERROR: AssertionError: isone(fN.unit)
Stacktrace:
 [1] _factor_assume_squarefree_and_separable(f::AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.FunctionFieldElem{QQFieldElem}})
   @ Hecke.FactorFF ~/.julia/packages/Hecke/VN62X/src/FunField/Factor.jl:224
 [2] _factor_assume_separable(f::AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.FunctionFieldElem{QQFieldElem}})
   @ Hecke.FactorFF ~/.julia/packages/Hecke/VN62X/src/FunField/Factor.jl:194
 [3] factor(f::AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.FunctionFieldElem{QQFieldElem}})
   @ Hecke.FactorFF ~/.julia/packages/Hecke/VN62X/src/FunField/Factor.jl:185
 [4] roots(f::AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.FunctionFieldElem{QQFieldElem}})
   @ AbstractAlgebra.Generic ~/.julia/packages/AbstractAlgebra/XwOtf/src/generic/Misc/Poly.jl:58
 [5] roots(R::AbstractAlgebra.Generic.FunctionField{QQFieldElem}, f::AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.RationalFunctionFieldElem{QQFieldElem, QQPolyRingElem}})
   @ AbstractAlgebra.Generic ~/.julia/packages/AbstractAlgebra/XwOtf/src/generic/Misc/Poly.jl:77
 [6] top-level scope
   @ REPL[8]:1
```
Further information on this issue (I don't know whether they might be helpful):
- The particular choice of `g` in this example can be obtained from `f` as `_, a = function_field(f); minpoly(leading_coefficient(f) * a)`
- This issue is independent of https://github.com/Nemocas/AbstractAlgebra.jl/pull/2393/.

--- a/src/FunField/Factor.jl
+++ b/src/FunField/Factor.jl
@@ -190,6 +190,7 @@ end
 
 #plain vanilla Trager, possibly doomed in pos. small char.
 function _factor_assume_squarefree_and_separable(f::Generic.Poly{<:Generic.AbsSimpleFunctionFieldElem})
+  @assert is_monic(f)
   i = 0
   local N
   g = f
@@ -211,7 +212,7 @@ function _factor_assume_squarefree_and_separable(f::Generic.Poly{<:Generic.AbsSi
   end
 
   fN = factor(N)
-  @assert isone(fN.unit)
+  # We are reconstructing the monic polynomial g and the gcds below are monic.
   D = Fac(one(parent(f)), Dict((gcd(map_coefficients(base_ring(f), p, parent = parent(f)), g)(t+i*a), k) for (p,k) = fN.fac))
   return D
 end

--- a/test/FunField/Factor.jl
+++ b/test/FunField/Factor.jl
@@ -12,6 +12,16 @@
   @test unit(fac) * prod(p^e for (p, e) in fac) == f
   @test all(isone(degree(p)) for (p, _) in fac)
 
+  # The factors of the norm need not be monic over a rational function field.
+  Qt, = rational_function_field(QQ, :t)
+  Qtx, x = Qt[:x]
+  K, a = function_field(x^2 + 1//2*x + 1, :a)
+  Ky, y = K[:y]
+  f = y - a
+  fac = factor(f)
+  @test evaluate(fac) == f
+  @test all(isone(degree(p)) for (p, _) in fac)
+
   # inseparable extension
   let
     k, o = finite_field(9)

